server/src/main/java/org/candlepin/dto/api/v1/NestedEntitlementTranslator.java

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright (c) 2009 - 2017 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.dto.api.v1;
+
+import org.candlepin.dto.ModelTranslator;
+import org.candlepin.dto.ObjectTranslator;
+import org.candlepin.model.Entitlement;
+import org.candlepin.util.Util;
+
+/**
+ * The EntitlementTranslator provides translation from Entitlement model objects to EntitlementDTOs.
+ */
+public class NestedEntitlementTranslator implements ObjectTranslator<Entitlement, NestedEntitlementDTO> {
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public NestedEntitlementDTO translate(Entitlement source) {
+        return this.translate(null, source);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public NestedEntitlementDTO translate(ModelTranslator translator, Entitlement source) {
+        return source != null ? this.populate(translator, source, new NestedEntitlementDTO()) : null;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public NestedEntitlementDTO populate(Entitlement source, NestedEntitlementDTO destination) {
+        return this.populate(null, source, destination);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public NestedEntitlementDTO populate(
+        ModelTranslator modelTranslator, Entitlement source, NestedEntitlementDTO dest) {
+        if (source == null) {
+            throw new IllegalArgumentException("source is null");
+        }
+
+        if (dest == null) {
+            throw new IllegalArgumentException("destination is null");
+        }
+
+        dest.setId(source.getId());
+        dest.setQuantity(source.getQuantity());
+        dest.setDeletedFromPool(source.deletedFromPool());
+        dest.setStartDate(Util.toDateTime(source.getStartDate()));
+        dest.setEndDate(Util.toDateTime(source.getEndDate()));
```

Comment:
Technically it is correct. Since we follow fluent style in most of our translators, maybe we can do it here as well.

Reply:
Done.